src/player.cpp

```diff
@@ -12401,48 +12401,54 @@ void get_armor_on(player* p, body_part bp, std::vector<int>& armor_indices) {
 }
 
 void player::armor_absorb(damage_unit& du, item& armor) {
-    if (rng(0,100) <= armor.get_coverage()) {
-        if (armor.is_power_armor()) { // TODO: add some check for power armor
-        }
-
-        const float effective_resist = resistances(armor).get_effective_resist(du);
-        // Amount of damage mitigated
-        const float mitigation = std::min(effective_resist, du.amount);
-        du.amount -= mitigation; // mitigate the damage first
-
-        // Scale chance of article taking damage based on the number of parts it covers.
-        // This represents large articles being able to take more punishment
-        // before becoming inneffective or being destroyed.
-        const int num_parts_covered = armor.get_covered_body_parts().count();
-        if( !one_in( num_parts_covered ) ) {
-            return;
-        }
+    if (rng(0,100) > armor.get_coverage()) {
```

Comment:
Shouldn't this be a `>=`? Coverage of 0 should give no change (currently it has a 1% chance).

And it should probably be rng(0,99) because otherwise (with `>=`) even 100 would not grant complete protection.

Or is this all by design and works as intended?